components/aws/sagemaker/batch_transform/src/batch_transform.py

```diff
@@ -13,27 +13,47 @@
+  parser.add_argument('--max_payload', type=_utils.str_to_int, required=False, help='The maximum allowed size of the payload, in MB.', default='6')
+  parser.add_argument('--batch_strategy', choices=['MultiRecord', 'SingleRecord', ''], required=False, help='The number of records to include in a mini-batch for an HTTP inference request.', default='')
+  parser.add_argument('--environment', type=_utils.str_to_json_dict, required=False, help='The dictionary of the environment variables to set in the Docker container. Up to 16 key-value entries in the map.', default='{}')
+  parser.add_argument('--input_location', type=str, required=True, help='The S3 location of the data source that is associated with a channel.')
+  parser.add_argument('--data_type', choices=['ManifestFile', 'S3Prefix', 'AugmentedManifestFile', ''], required=False, help='Data type of the input. Can be ManifestFile, S3Prefix, or AugmentedManifestFile.', default='S3Prefix')
+  parser.add_argument('--content_type', type=str, required=False, help='The multipurpose internet mail extension (MIME) type of the data.', default='')
+  parser.add_argument('--split_type', choices=['None', 'Line', 'RecordIO', 'TFRecord', ''], required=False, help='The method to use to split the transform job data files into smaller batches.', default='None')
+  parser.add_argument('--compression_type', choices=['None', 'Gzip', ''], required=False, help='If the transform data is compressed, the specification of the compression type.', default='None')
+  parser.add_argument('--output_location', type=str, required=True, help='The Amazon S3 path where you want Amazon SageMaker to store the results of the transform job.')
+  parser.add_argument('--accept', type=str, required=False, help='The MIME type used to specify the output data.')
+  parser.add_argument('--assemble_with', choices=['None', 'Line', ''], required=False, help='Defines how to assemble the results of the transform job as a single S3 object. Either None or Line.')
+  parser.add_argument('--output_encryption_key', type=str, required=False, help='The AWS KMS key that Amazon SageMaker uses to encrypt the model artifacts.', default='')
+  parser.add_argument('--input_filter', type=str, required=False, help='A JSONPath expression used to select a portion of the input data to pass to the algorithm.', default='')
+  parser.add_argument('--output_filter', type=str, required=False, help='A JSONPath expression used to select a portion of the joined dataset to save in the output file for a batch transform job.', default='')
+  parser.add_argument('--join_source', choices=['None', 'Input', ''], required=False, help='Specifies the source of the data to join with the transformed data.', default='None')
+  parser.add_argument('--instance_type', choices=['ml.m4.xlarge', 'ml.m4.2xlarge', 'ml.m4.4xlarge', 'ml.m4.10xlarge', 'ml.m4.16xlarge', 'ml.m5.large', 'ml.m5.xlarge', 'ml.m5.2xlarge', 'ml.m5.4xlarge',
```

Comment:
Minor: We need to consider how to better maintain this list. It could easily out of sync once SM rollout new instances

Reply:
Is there a way to fetch supported instances using sagemaker api? like the way you get built-in algorithm?

Reply:
Will look into it or ask around, but there doesn't seem to be a way from a quick search through the sagemaker documentation